tests/validate/test_facades.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2023 Canonical Ltd.
+# Licensed under the Apache V2, see LICENCE file for details.
+
+import importlib
+import re
+from pathlib import Path
+from typing import Dict, List, TypedDict, cast
+
+import pytest
+
+from juju.client import connection, _definitions
+
+
+class Versions(TypedDict, total=True):
+    versions: List[int]
+
+
+ClientFacades = Dict[str, Versions]
+
+
+@pytest.fixture
+def project_root(pytestconfig: pytest.Config) -> Path:
+    return pytestconfig.rootpath
+
+
+def test_facade_version_matches_filename(project_root: Path) -> None:
+    ignore_names = dir(_definitions)
+    for file in (project_root / 'juju' / 'client').glob('_client[0-9]*.py'):
+        match = re.search('_client([0-9]+).py', file.name)
+        assert match
+        version = int(match.group(1))
+        module = importlib.import_module(f'juju.client.{file.stem}')
+        for cls_name in dir(module):
+            if cls_name.startswith('_') or cls_name in ignore_names:
+                continue
+            assert getattr(module, cls_name).version == version
+
+
+def test_class_name_matches_facade_name(project_root: Path) -> None:
+    ignore_names = dir(_definitions)
+    for file in (project_root / 'juju' / 'client').glob('_client[0-9]*.py'):
+        module = importlib.import_module(f'juju.client.{file.stem}')
+        for cls_name in dir(module):
+            if cls_name.startswith('_') or cls_name in ignore_names:
+                continue
+            assert getattr(module, cls_name).name == cls_name.removesuffix('Facade')
```

Comment:
Is there a mismatch precedent?

If there is not, let's remove these tests.

Reply:
Yeah I'm glad I checked that it matches, but it should by definition, so I'll remove these.